### ISSUE TITLE
Donations feature

### ISSUE THREAD

**Issue by hoyes** [camdram/camdram#298]
I briefly mentioned this in an e-mail to webteam@ and it didn't receive any opposition, so creating this to discuss the idea further.

For those not already aware, Camdram has some minimal (but not negligible) running costs that in the last few years have been generously donated by a small number of people. Going forwards, we would like Camdram to be financially self-sustainable.

The idea is for Camdram to accept donations (via Paypal or similar service). This has the advantage of allowing us to accept money without the bureaucracy that comes with having a bank account. The idea is that money is transferred out to people as necessary. 

This could operate in various ways, e.g.:

 a) Just a donations page (probably linked from the top nav bar next to 'About Us', 'Development' etc), from which people can make one-off donations.
 b) The above, plus sending annual e-mails to societies/venues to suggest they make a donation
 c) Require societies/venues that wish to use admin facilities on Camdram to pay an annual subscription.

My personal feeling would be just do a) to start off with (as it's the easiest to implement). Long term though, I wonder whether c) might be a viable option (obviously after consulting the various societies/venues to see what they thought). I know Camdram's outgoings haven't been fully confirmed yet, but I suspect that if every organisation that currently has admin access on Camdram paid just £10 a year then that would more than cover the operating costs.

I see this as being post-v1 switch-off in the list of priorities, but just starting to the discussion...

**Comment by philosophicles** [comment]
I agree with trying option a) first. If it proves sufficient, great. If not, I'd advocate trying option b) before c), if necessary with fairly direct wording "recommended donation of £X per society/venue". 

I personally don't like the sound of mandatory charges, because some societies operate essentially for free themselves (or at least did in my student era) and I wouldn't want to prevent socs from using Camdram if they really can't afford it.

**Comment by philosophicles** [comment]
As of mid-2018, I think I'm right that we have zero _actual_ running costs? (Since SSL certs are now free.)

I think a **wholly-voluntary** donation system would still be worthwhile - but the stated purpose of it (which should be transparent) might be a little different now. Personally, I'd see it as being: 

1. A way for users to 'tip' our web hosting provider, Loho - who does generously provide the hosting completely free, but is also a small business with costs to meet, and who does spend time working on Camdram in our capacity as their hosting client as well as choosing to do things for Camdram as a personal hobby.

2. A fund that could be used towards Camdram socials in Cambridge - which might allow Camdram volunteers/devs to get a free beer every so often for their hard work, but also might encourage other volunteers out of the woodwork, benefitting the whole operation. 

It might be nice to allow website donors to choose which of these things they want to donate towards, if they're agreed to both be worthwhile.

**Comment by philosophicles** [comment]
Interested to hear opinions of our newer devs @GKFX @CHTJonas on this one, too.

**Comment by CHTJonas** [comment]
I think that this is an important issue to consider so thanks for the bump @philosophicles!

Related to #21 if we're going down hosting the wiki route, maybe in asking the larger societies for donations we could increase the storage space we have on Tiberius? Now I have no idea how expensive it is to provision say another 30GB of storage for wiki file uploads but might be worth a discussion (maybe on webteam once wiki details are fleshed out & not on here though).

Otherwise we could definitely accept pay-pal donations and then these could be earmarked for either Loho or a Camdram community pot which we could use at hypothetical Camdram socials ~twice a year to buy a round of drinks for people interested in Camdram development/support/administration (ourselves perhaps included).

The only potential concerns I can think of at the moment (I don't think they're big issue but I'll mention just for the sake of it) are:

1. Political implications (ie. societies who can afford to donate seen to have more influence over Camdram)
2. Societies making donations might expect quicker development cycle/value for money. Obviously Camdram is developed as a hobby and has absolutely no SLA so we can't guarantee that feature X or bug Y will be added/fixed in the next 2/3 days.

**Comment by CHTJonas** [comment]
What do people think is the best method of accepting donations e.g. open a Camdram bank account, PayPal, etc? Depending on the route we take there may be some involved bureaucratic process.

Does anyone have any first-hand experience of setting something like this up?
